drivers/adc/adc_shell.c

```diff
@@ -0,0 +1,420 @@
+	if (chosen_adc < 0) {
+		return -EINVAL;
+	}
+
+	adc_dev = device_get_binding(adc_list[chosen_adc].device_name);
+	if (adc_dev == NULL) {
+		return -ENODEV;
+	}
+	for (i = 0; i < ARRAY_SIZE(reference_list); i++) {
+		if (!strcmp(argv[args_indx.conf], reference_list[i].string)) {
+			adc_list[chosen_adc].channel_config.reference =
+				reference_list[i].reference;
+			retval = adc_channel_setup(adc_dev,
+					&adc_list[chosen_adc].channel_config);
+			LOG_DBG("Channel setup returned %i\n", retval);
+			break;
+		}
+	}
+	retval = adc_channel_setup(adc_dev, &adc_list[chosen_adc].channel_config);
+	return retval;
+}
+#define BUFFER_SIZE 1
+static int cmd_adc_read(const struct shell *shell, size_t argc, char **argv)
+{
+	int retval = 0;
+	int chosen_adc = -1;
+	struct device *adc_dev;
+	u16_t m_sample_buffer[BUFFER_SIZE];
+
+	if (argc != args_no.read) {
+		shell_fprintf(shell, SHELL_NORMAL,
+				"Usage: read <adc> <channel>\n");
+		return 0;
+	}
+	chosen_adc = get_adc_from_list(argv[args_indx.adc]);
+	if (chosen_adc < 0) {
+		return 0;
+	}
+	u8_t adc_channel_id = strtol(argv[args_indx.channel], NULL, 10);
+
+	adc_dev = device_get_binding(adc_list[chosen_adc].device_name);
+	if (adc_dev == NULL) {
+		return -ENODEV;
+	}
+	adc_list[chosen_adc].channel_config.channel_id = adc_channel_id;
+	const struct adc_sequence sequence = {
+		.channels	=
+			BIT(adc_list[chosen_adc].channel_config.channel_id),
+		.buffer		= m_sample_buffer,
+		.buffer_size	= sizeof(m_sample_buffer),
+		.resolution	= adc_list[chosen_adc].resolution,
+	};
+	retval = adc_read(adc_dev, &sequence);
+	if (retval >= 0) {
+		shell_fprintf(shell, SHELL_NORMAL,
+				"Read: %i\n", m_sample_buffer[0]);
+	}
+	return retval;
+}
+
+static int cmd_adc_print(const struct shell *shell, size_t argc, char **argv)
+{
+	int chosen_adc = -1;
+	int i;
+	char *gain = "1";
+	char *ref = "INTERNAL";
+	u16_t acq_time;
+	u8_t channel_id;
+	u8_t resolution;
+
+	if (argc < 2) {
+		shell_fprintf(shell, SHELL_NORMAL, "usage: print <adc>");
+	}
+	chosen_adc = get_adc_from_list(argv[args_indx.adc]);
+	if (chosen_adc < 0) {
+		return 0;
+	}
+	for (i = 0; i < ARRAY_SIZE(gain_list); i++) {
+		if (gain_list[i].gain ==
+				adc_list[chosen_adc].channel_config.gain) {
+			gain = gain_list[i].string;
+		}
+	}
+	for (i = 0; i < ARRAY_SIZE(reference_list); i++) {
+		if (reference_list[i].reference ==
+				adc_list[chosen_adc].channel_config.reference) {
+			ref = reference_list[i].string;
+		}
+	}
+	acq_time = adc_list[chosen_adc].channel_config.acquisition_time;
+	channel_id = adc_list[chosen_adc].channel_config.channel_id;
+	resolution = adc_list[chosen_adc].resolution;
+	shell_fprintf(shell, SHELL_NORMAL, "%s:\n"
+			"Gain: %s\n"
+			"Reference: %s\n"
+			"Acquisition Time: %u\n"
+			"Channel ID: %u\n"
+			"Resolution: %u\n",
+			argv[args_indx.adc],
+			gain,
+			ref,
+			acq_time,
+			channel_id,
+			resolution);
+	return 0;
+}
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_adc) {
```

Comment:
The `SHELL_CREATE_STATIC_SUBCMD_SET` macro is deprecated. Use `SHELL_STATIC_SUBCMD_SET_CREATE` instead.